src/plugin/class-promoter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace DotOrg\TryWordPress;
+
+use WP_Post;
+
+class Promoter {
+	private string $meta_key_for_promoted_post = '_promoted_post';
+
+	public function __construct() {}
+
+	private function get_post_type_for_promoted_post( string $post_type ): string {
+		return str_replace( 'liberated_', '', $post_type );
+	}
+
+	public function get_promoted_post( $liberated_post_id ): int {
+		return absint( get_post_meta( $liberated_post_id, $this->meta_key_for_promoted_post, true ) );
```

Comment:
From what I understood from the `get_post_meta()` docs, it will return an empty string when the post (or the meta) doesn't exist. I'm not sure what `absint()` returns when it takes an empty string.

I think maybe we should make the return value of this function `int | null` and return null when the meta is empty.

Reply:
Fair point! I will choose strict types from now on. WordPress is quite inconsistent even with truthy or falsy values. No point in a single return type if it's still truth or falsy, as I did here.

In this example: it would have been zero or non-zero value because of `absint`. So, `if` would be enough to catch truthy or falsy values. But semantically still wrong. `null` is better to indicate absence of value than a falsy value. 👍